client/ayon_maya/plugins/publish/extract_model.py

```diff
@@ -76,9 +77,12 @@ def process(self, instance):
                                        pointsWire=4,
                                        pointsShaded=1,
                                        polygonObject=1):
-                with lib.shader(members,
-                                shadingEngine="initialShadingGroup"):
-                    with lib.maintained_selection():
+                with lib.maintained_selection():
+                    with contextlib.ExitStack() as stack:
+                        if instance.data.get("writeFaceSets", True):
+                            stack.enter_context(
+                                lib.write_face_sets_for_materials(members)
+                            )
```

Comment:
If we are using a `ExitStack` now I think it's a good time to just move all the context managers into the `ExitStack`.
That would also avoid the deep nesting we have now.